### PR TITLE
Extend exception-related checks to `except*`

### LINKED ISSUE #8827: Extend broad-exception-raised and broad-exception-caught to `except*`
_Originally posted by @clavedeluna in https://github.com/pylint-dev/pylint/issues/7703#issuecomment-1303843158_

***
Expect these examples to raise the same message:

```python
try:
    1 / 0
except Exception:
    pass
```

************* Module a
a.py:3:7: W0718: Catching too general exception Exception (broad-exception-caught)

```python
try:
    1 / 0
except* Exception:
    pass
```

************* Module a
a.py:2:4: W0104: Statement seems to have no effect (pointless-statement)

### PR DESCRIPTION
**Pull request by azinneck0485**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Modify exceptions.py checks to detect `try-except*` blocks and treat the same as `try-except`.

Also updated basic_checker.py so that `try-except*` blocks are not flagged as "pointless-statement"

Functional tests for `broad-exception-caught` and `broad-exception-raised` were updated to include `try-except*`.  All tests passed locally with `pytest`, `tox`, and `primer`.

Closes #8827

### CODE REVIEW

File: pylint/checkers/base/basic_checker.py [hunk 1]
```diff
@@ -479,6 +479,7 @@ def visit_expr(self, node: nodes.Expr) -> None:
         if (
             isinstance(expr, (nodes.Yield, nodes.Await))
             or (isinstance(node.parent, nodes.Try) and node.parent.body == [node])
+            or (isinstance(node.parent, nodes.TryStar) and node.parent.body == [node])
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/base/basic_checker.py:482]:
```suggestion
            or (isinstance(node.parent, (nodes.Try, nodes.TryStar) and node.parent.body == [node])
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
I have a nit but this is going straight into 3.1.0, great work :)

**Comment by azinneck0485** [comment]
thank you!

and thanks for the suggested change, I wasn't aware of that syntax and it looks so much better with less repeated code :-)

File: pylint/checkers/base/basic_checker.py [hunk 2]
```diff
@@ -478,7 +478,7 @@ def visit_expr(self, node: nodes.Expr) -> None:
         # side effects), else pointless-statement
         if (
             isinstance(expr, (nodes.Yield, nodes.Await))
-            or (isinstance(node.parent, nodes.Try) and node.parent.body == [node])
+            or (isinstance(node.parent, (nodes.Try, nodes.TryStar) and node.parent.body == [node])
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/base/basic_checker.py:481]:
```suggestion
            or (isinstance(node.parent, (nodes.Try, nodes.TryStar)) and node.parent.body == [node])
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for your contribution, congrats on becoming a pylint contributor 🎉

**Pull request merged**